GeneralsMD/Code/GameEngineDevice/Source/W3DDevice/GameClient/W3DSmudge.cpp

```diff
@@ -535,9 +539,10 @@ void W3DSmudgeManager::render(RenderInfoClass &rinfo)
 				if (set)	//start next batch at beginning of set.
 					remainingSmudgeStart = set->getUsedSmudgeList().Head();
 			}
-flushSmudges:
-			DX8Wrapper::Set_Vertex_Buffer(vb_access);
 		}
+flushSmudges:
+		++smudgesBatchCount;
+		DX8Wrapper::Set_Vertex_Buffer(vb_access);
```

Comment:
I moved this call outside the above scope, because that was within the lock of the vertex buffer, which did not appear to cause issues but is technically not right.